test/validation/api/dma/dma.c

```diff
@@ -364,6 +366,38 @@ static void test_dma_compl_pool_max_pools(void)
 	}
 }
 
+static void test_dma_compl_user_area(void)
+{
+	odp_dma_pool_param_t dma_pool_param;
+	uint32_t size = global.dma_capa.pool.max_uarea_size;
+	odp_pool_t pool;
+	odp_dma_compl_t compl_ev;
+
+	if (size == 0) {
+		ODPH_DBG("DMA completion event user area not supported\n");
+		return;
+	}
+
+	odp_dma_pool_param_init(&dma_pool_param);
+	dma_pool_param.num = 1;
+	dma_pool_param.uarea_size = size;
+	pool = odp_dma_pool_create(NULL, &dma_pool_param);
+
+	CU_ASSERT_FATAL(pool != ODP_POOL_INVALID);
+
+	compl_ev = odp_dma_compl_alloc(pool);
+
+	CU_ASSERT(compl_ev != ODP_DMA_COMPL_INVALID);
+
+	if (compl_ev != ODP_DMA_COMPL_INVALID) {
+		CU_ASSERT(odp_dma_compl_user_area(compl_ev) != NULL);
+
+		odp_dma_compl_free(compl_ev);
+	}
```

Comment:
Perhaps a simple loop test could be added, which allocates a number of completion events, checks that user area is different per event, and fills user area with  dummy data.

Reply:
Refactored to case to be very similar to other user area tests (buffer, timer, etc.).